db/app.js

```diff
@@ -3,19 +3,33 @@ const app = express();
 const {
   searchTopics,
   searchEndpoints,
+  searchArticleById,
 } = require("../db/controller/app.controller");
 
 app.get("/api/topics", searchTopics);
 
 app.get("/api", searchEndpoints);
 
+app.get("/api/articles/:article_id", searchArticleById);
+
 app.all("*", (_, res) => {
   res.status(404).send({ status: 404, msg: "Not found" });
 });
 
 app.use((err, req, res, next) => {
-  return res.status(500).send({ mgs: "Internal Server Error" });
+  if (err.msg) {
+    res.status(404).send({ msg: err.msg });
```

Comment:
This works fine here so doesn't need to change but if you had a custom error that needed a dynamic status code, you are already passing the code on the `thrown` error object so you can access it there